org.eclipse.gef/src/org/eclipse/gef/editpolicies/SelectionHandlesEditPolicy.java

```diff
@@ -51,38 +51,33 @@ protected void addSelectionHandles() {
 		removeSelectionHandles();
 		IFigure layer = getLayer(LayerConstants.HANDLE_LAYER);
 		handles = createSelectionHandles();
-		for (Object handle : handles) {
-			layer.add((IFigure) handle);
-		}
+		handles.stream().filter(IFigure.class::isInstance).map(IFigure.class::cast).forEach(layer::add);
```

Comment:
Would it make sense to have a private helper function to get the Handle figures? The return type should then probably be a stream...